### ISSUE TITLE
[Bug]: Downloading the iCla/ics file does not work on iOS

### ISSUE THREAD

**Issue by mrc-bsllt** [add2cal/add-to-calendar-button#322]
### What happened?

On iOS devices, when I click on the button to download the event file, the message 'Add to Calendar Button "atcb-btn-custom" triggered' appears in the console, but the file is not downloaded. Instead, the buttons to add the event to the calendar (Google, Yahoo, etc.) work. Everything works on Android too!
On Android devices, when I download the event, there is an additional message in the console, saying 'Event saved. Looking forward to it!"

### System & Environment

I'm using:
- Nuxt 2.15.18;
- add-to-calendar v1.18.8;
- Iphone 11 and Iphone 13

**Comment by jekuer** [comment]
Hi @mrc-bsllt , can you try out the new v2 of the script?
Since a lot changed, it is hard to say whether your issue has been resolved in the meantime or not.
Thanks.

**Comment by mrc-bsllt** [comment]
> Hi @mrc-bsllt , can you try out the new v2 of the script? Since a lot changed, it is hard to say whether your issue has been resolved in the meantime or not. Thanks.

Hello @jekuer and thanks for your reply.
With the last version i've got this error: 
**"require() of ES Module .../node_modules/add-to-calendar-button/dist/module/index.js from .../node_modules/vue-server-renderer/build.dev.js not supported. Instead change the require of index.js in .../node_modules/vue-server-renderer/build.dev.js to a dynamic import() which is available in all CommonJS modules."**

**Comment by jekuer** [comment]
How do you import the script?

Are you using "require ..." or "import 'add-to-calendar-button';"

**Comment by mrc-bsllt** [comment]
> How do you import the script?
> 
> Are you using "require ..." or "import 'add-to-calendar-button';"

like this:
```javascript
<script>
import { atcb_action as atcbAction } from 'add-to-calendar-button'

...
</script>
```

**Comment by jekuer** [comment]
One last question: Are you using SSR or SSG?
Seems like we need to re-add commonJS support.

**Comment by mrc-bsllt** [comment]
> One last question: Are you using SSR or SSG? Seems like we need to re-add commonJS support.

I'm using SSR.
Thanks again for your reply!

**Comment by add-to-calendar** [comment]
For the latest version, you should add the following to your nuxt.config.js:
```
build: {
  transpile: ['add-to-calendar-button'],
},
```

Hope this also solves your main issue 🤞

**Comment by mrc-bsllt** [comment]
> transpile: ['add-to-calendar-button'],

Hi, I added the transpile in my nuxt.config.js and the import works, but now my feature no longer works. There are some differences from before. In my case I hide the whole widget and trigger a click event in the button I select by id. Now, compared to before, the html structure that is generated is different and no longer lets me do this. I attach an image of the before and after:

Before:
![before](https://user-images.githubusercontent.com/72490444/215999269-747d603f-5610-47b4-aba1-96cedd39a2ec.png)

After:
![after](https://user-images.githubusercontent.com/72490444/215999333-bc28a228-a06b-4e37-9e33-5f0ec8489838.png)

I also enclose my code so you understand how I am handling this feature:

```javascript
<template>
  <div class="flex flex-col justify-start items-center flex-nowrap mt-4 mb-2 py-3 px-4 bg-gray-3 rounded-[8px]">
    <div class="w-full flex flex-row justify-center items-center flex-nowrap">
      <the-icon name="calendar" width="15" height="15" class="mr-2" />
      <span class="text-body-normal">
        {{ $t('funnel.pages.booking_confirmation.add_to_calendar') }}
      </span>
    </div>
    <ul class="w-full flex flex-row justify-center items-center flex-wrap mt-1">
        <li
          v-for="(prov, index) in providers"
          :key="index"
          class="flex flex-row justify-start items-center flex-nowrap"
        >
          <button 
            type="button" 
            class="text-body-normal font-bold underline"
            @click="($event) => addToCalendar($event, index)"
          >
            {{ prov.label }}
          </button>
          <the-icon 
            v-if="index < providers.length - 1"
            name="point" 
            width="4" 
            height="4" 
            class="shrink-0 mx-1" 
          />
        </li>
    </ul>
  </div>
</template>

<script>
import { atcb_action as atcbAction } from 'add-to-calendar-button'
import TheIcon from '@/components/commons/TheIcon.vue'

export default {
  name: 'AddToCalendarButtons',
  components: {
    TheIcon,
  },
  props: {
    date: {
      type: String,
      required: true
    },
    pnr: {
      type: String,
      required: true
    },
    visitors: {
      type: Array,
      required: true
    }
  },
  data() {
    return {
      providers: [
        {
          label: 'Google',
          option: 'Google',
          containerId: 'atcb-btn-custom-google'
        },
        {
          label: 'Yahoo',
          option: 'Yahoo',
          containerId: 'atcb-btn-custom-yahoo'
        },
        {
          label: 'Outlook',
          option: 'Outlook.com',
          containerId: 'atcb-btn-custom-outlookcom'
        },
        {
          label: 'Scarica evento',
          option: 'iCal',
          containerId: 'atcb-btn-custom-ical'
        }
      ]
    }
  },
  computed: {
    visitorsForEmailDescription() {
      return this.visitors
        .map(v => `\u00A0\u00A0- ${v.nome} ${v.cognome}[br]`)
        .join('')
    },
    config() {
      return {
         ...myConfig
      }
    }
  },
  methods: {
    addToCalendar($event, index) {
      const button = $event.target
      atcbAction(this.config, button)

      const id = this.providers[index].containerId
      const hiddenButton = document.getElementById(id)
      hiddenButton.click()
    }
  }
}
</script>

<style lang="scss">
#atcb-pos-wrapper {
  @apply w-[50px] h-[50px] opacity-0 #{!important};

  svg {
    @apply w-[50px] h-[50px] #{!important};
  }
}
</style>
```

**Comment by mrc-bsllt** [comment]
> > transpile: ['add-to-calendar-button'],
> 
> Hi, I added the transpile in my nuxt.config.js and the import works, but now my feature no longer works. There are some differences from before. In my case I hide the whole widget and trigger a click event in the button I select by id. Now, compared to before, the html structure that is generated is different and no longer lets me do this. I attach an image of the before and after:
> 
> Before: ![before](https://user-images.githubusercontent.com/72490444/215999269-747d603f-5610-47b4-aba1-96cedd39a2ec.png)
> 
> After: ![after](https://user-images.githubusercontent.com/72490444/215999333-bc28a228-a06b-4e37-9e33-5f0ec8489838.png)
> 
> I also enclose my code so you understand how I am handling this feature:
> 
> ```js
> <template>
>   <div class="flex flex-col justify-start items-center flex-nowrap mt-4 mb-2 py-3 px-4 bg-gray-3 rounded-[8px]">
>     <div class="w-full flex flex-row justify-center items-center flex-nowrap">
>       <the-icon name="calendar" width="15" height="15" class="mr-2" />
>       <span class="text-body-normal">
>         {{ $t('funnel.pages.booking_confirmation.add_to_calendar') }}
>       </span>
>     </div>
>     <ul class="w-full flex flex-row justify-center items-center flex-wrap mt-1">
>         <li
>           v-for="(prov, index) in providers"
>           :key="index"
>           class="flex flex-row justify-start items-center flex-nowrap"
>         >
>           <button 
>             type="button" 
>             class="text-body-normal font-bold underline"
>             @click="($event) => addToCalendar($event, index)"
>           >
>             {{ prov.label }}
>           </button>
>           <the-icon 
>             v-if="index < providers.length - 1"
>             name="point" 
>             width="4" 
>             height="4" 
>             class="shrink-0 mx-1" 
>           />
>         </li>
>     </ul>
>   </div>
> </template>
> 
> <script>
> import { atcb_action as atcbAction } from 'add-to-calendar-button'
> import TheIcon from '@/components/commons/TheIcon.vue'
> 
> export default {
>   name: 'AddToCalendarButtons',
>   components: {
>     TheIcon,
>   },
>   props: {
>     date: {
>       type: String,
>       required: true
>     },
>     pnr: {
>       type: String,
>       required: true
>     },
>     visitors: {
>       type: Array,
>       required: true
>     }
>   },
>   data() {
>     return {
>       providers: [
>         {
>           label: 'Google',
>           option: 'Google',
>           containerId: 'atcb-btn-custom-google'
>         },
>         {
>           label: 'Yahoo',
>           option: 'Yahoo',
>           containerId: 'atcb-btn-custom-yahoo'
>         },
>         {
>           label: 'Outlook',
>           option: 'Outlook.com',
>           containerId: 'atcb-btn-custom-outlookcom'
>         },
>         {
>           label: 'Scarica evento',
>           option: 'iCal',
>           containerId: 'atcb-btn-custom-ical'
>         }
>       ]
>     }
>   },
>   computed: {
>     visitorsForEmailDescription() {
>       return this.visitors
>         .map(v => `\u00A0\u00A0- ${v.nome} ${v.cognome}[br]`)
>         .join('')
>     },
>     config() {
>       return {
>          ...myConfig
>       }
>     }
>   },
>   methods: {
>     addToCalendar($event, index) {
>       const button = $event.target
>       atcbAction(this.config, button)
> 
>       const id = this.providers[index].containerId
>       const hiddenButton = document.getElementById(id)
>       hiddenButton.click()
>     }
>   }
> }
> </script>
> 
> <style lang="scss">
> #atcb-pos-wrapper {
>   @apply w-[50px] h-[50px] opacity-0 #{!important};
> 
>   svg {
>     @apply w-[50px] h-[50px] #{!important};
>   }
> }
> </style>
> ```

I can obviously select the elements anyway, but the fact that the ids are 'undefined-' makes me think of something unintended.

**Comment by mrc-bsllt** [comment]
> > > transpile: ['add-to-calendar-button'],
> > 
> > 
> > Hi, I added the transpile in my nuxt.config.js and the import works, but now my feature no longer works. There are some differences from before. In my case I hide the whole widget and trigger a click event in the button I select by id. Now, compared to before, the html structure that is generated is different and no longer lets me do this. I attach an image of the before and after:
> > Before: ![before](https://user-images.githubusercontent.com/72490444/215999269-747d603f-5610-47b4-aba1-96cedd39a2ec.png)
> > After: ![after](https://user-images.githubusercontent.com/72490444/215999333-bc28a228-a06b-4e37-9e33-5f0ec8489838.png)
> > I also enclose my code so you understand how I am handling this feature:
> > ```js
> > <template>
> >   <div class="flex flex-col justify-start items-center flex-nowrap mt-4 mb-2 py-3 px-4 bg-gray-3 rounded-[8px]">
> >     <div class="w-full flex flex-row justify-center items-center flex-nowrap">
> >       <the-icon name="calendar" width="15" height="15" class="mr-2" />
> >       <span class="text-body-normal">
> >         {{ $t('funnel.pages.booking_confirmation.add_to_calendar') }}
> >       </span>
> >     </div>
> >     <ul class="w-full flex flex-row justify-center items-center flex-wrap mt-1">
> >         <li
> >           v-for="(prov, index) in providers"
> >           :key="index"
> >           class="flex flex-row justify-start items-center flex-nowrap"
> >         >
> >           <button 
> >             type="button" 
> >             class="text-body-normal font-bold underline"
> >             @click="($event) => addToCalendar($event, index)"
> >           >
> >             {{ prov.label }}
> >           </button>
> >           <the-icon 
> >             v-if="index < providers.length - 1"
> >             name="point" 
> >             width="4" 
> >             height="4" 
> >             class="shrink-0 mx-1" 
> >           />
> >         </li>
> >     </ul>
> >   </div>
> > </template>
> > 
> > <script>
> > import { atcb_action as atcbAction } from 'add-to-calendar-button'
> > import TheIcon from '@/components/commons/TheIcon.vue'
> > 
> > export default {
> >   name: 'AddToCalendarButtons',
> >   components: {
> >     TheIcon,
> >   },
> >   props: {
> >     date: {
> >       type: String,
> >       required: true
> >     },
> >     pnr: {
> >       type: String,
> >       required: true
> >     },
> >     visitors: {
> >       type: Array,
> >       required: true
> >     }
> >   },
> >   data() {
> >     return {
> >       providers: [
> >         {
> >           label: 'Google',
> >           option: 'Google',
> >           containerId: 'atcb-btn-custom-google'
> >         },
> >         {
> >           label: 'Yahoo',
> >           option: 'Yahoo',
> >           containerId: 'atcb-btn-custom-yahoo'
> >         },
> >         {
> >           label: 'Outlook',
> >           option: 'Outlook.com',
> >           containerId: 'atcb-btn-custom-outlookcom'
> >         },
> >         {
> >           label: 'Scarica evento',
> >           option: 'iCal',
> >           containerId: 'atcb-btn-custom-ical'
> >         }
> >       ]
> >     }
> >   },
> >   computed: {
> >     visitorsForEmailDescription() {
> >       return this.visitors
> >         .map(v => `\u00A0\u00A0- ${v.nome} ${v.cognome}[br]`)
> >         .join('')
> >     },
> >     config() {
> >       return {
> >          ...myConfig
> >       }
> >     }
> >   },
> >   methods: {
> >     addToCalendar($event, index) {
> >       const button = $event.target
> >       atcbAction(this.config, button)
> > 
> >       const id = this.providers[index].containerId
> >       const hiddenButton = document.getElementById(id)
> >       hiddenButton.click()
> >     }
> >   }
> > }
> > </script>
> > 
> > <style lang="scss">
> > #atcb-pos-wrapper {
> >   @apply w-[50px] h-[50px] opacity-0 #{!important};
> > 
> >   svg {
> >     @apply w-[50px] h-[50px] #{!important};
> >   }
> > }
> > </style>
> > ```
> 
> I can obviously select the elements anyway, but the fact that the ids are 'undefined-' makes me think of something unintended.

Another strange thing I just noticed, is that if I put the responsive mode on, the iCal element to download the event disappears
![strange](https://user-images.githubusercontent.com/72490444/216008854-3e832dee-545b-43e5-ab95-840a01035923.png)

**Comment by jekuer** [comment]
Hi @mrc-bsllt ,
Thanks for your report. Made me uncover 2 major bugs.
1, which only occurs if you use the atcb_action with an button element, that has no id. "undefined" should now be gone. Btw, ideally, your triggering element holds a unique id. Optional, but makes it a little bit more stable.
2, the iCal gets replaced by the Apple option on iOS (if the Apple option has not been set anyway). This is, because iOS will handle those two options the exact same way. This process was broken.

Both fixes with latest v2.1.1
